climada/hazard/trop_cyclone.py

```diff
@@ -629,16 +629,39 @@ def _apply_knutson_criterion(
             # Scale frequencies by category
             cat_larger_list = []
             for chg in freq_chg:
+                # whole cat group
+                sel_cat_all = (np.isin(tc_cc.category, chg['category']) & bas_sel)
+
                 cat_chg_list = [cat
                                 for cat in chg['category']
                                 if cat not in cat_larger_list
                                 ]
+
+                # cats to change
                 sel_cat_chg = np.isin(tc_cc.category, cat_chg_list) & bas_sel
+
+                # cats already changed
+                sel_cat_larger = (np.isin(tc_cc.category, cat_larger_list) & bas_sel)
+
                 if sel_cat_chg.any():
-                    # prevent freq_scaling from becoming negative which would lead
-                    # to negative frequencies
-                    freq_scaling = np.max([1 + (chg['change'] - 1) * scaling_rcp_year, 0])
-                    tc_cc.frequency[sel_cat_chg] *= freq_scaling
+                    # scaling factor to whole cat group
+                    freq_scaling = 1 + (chg['change'] - 1) * scaling_rcp_year
+
+                    # scaling factor to missing cats to change
+                    freq_scaling_cor = (
+                        (np.sum(self.frequency[sel_cat_all]) * freq_scaling
+                         - np.sum(tc_cc.frequency[sel_cat_larger]))
+                        / np.sum(self.frequency[sel_cat_chg])
+                    )
+
+                    if freq_scaling_cor < 0:
+                        LOGGER.warning("The application of the given climate scenario"
+                                       " would result in negative frequency correction factors."
+                                       " Frequencies are thus left unchanged but this"
+                                       " should warrant attention from the user.")
+                        freq_scaling_cor = 1
```

Comment:
I am not sure that this is a good handling of this case. A user will simply not know what to do here. So, we should either throw an error or find an actual solution that is defendable. Is just not scaling the frequencies really a good solution?